opm/simulators/linalg/WellOperators.hpp

```diff
@@ -123,10 +123,46 @@ class WellModelAsLinearOperator : public LinearOperatorExtra<X, Y>
         return wellMod_.numLocalWellsEnd();
     }
 
-private:
+protected:
     const WellModel& wellMod_;
 };
 
+
+template <class WellModel, class X, class Y>
+class DomainWellModelAsLinearOperator : public WellModelAsLinearOperator<WellModel, X, Y>
+{
+public:
+    using Base = WellModelAsLinearOperator<WellModel, X, Y>;
+    using Base::Base;
```

Comment:
This seems both unnecessary and confusing.

Reply:
I agree that it seems confusing. The `using Base::Base;` is necessary for inheriting constructors from base class.
Open for other suggestions.

Reply:
Changed the naming and added a comment for more clarity.